core/src/main/java/org/mule/runtime/core/internal/execution/SourceResultAdapter.java

```diff
@@ -48,7 +50,7 @@ public SourceResultAdapter(Result<?, ?> result,
                              MediaType mediaType,
                              boolean isCollection,
                              Optional<String> correlationId) {
-    this(result, cursorProviderFactory, mediaType, isCollection, correlationId, null);
+    this(result, cursorProviderFactory, mediaType, isCollection, correlationId, null, null);
```

Comment:
why would this be optional? Avoid nulls

Reply:
because it will only be present when the message originated from a polling source

Reply:
then receive it as optional